packages/ckeditor5-image/src/imagestyle/imagestyleui.js

```diff
@@ -73,85 +75,96 @@ export default class ImageStyleUI extends Plugin {
 
 		const definedStyles = translateStyles(
-			normalizedStyles.options,
-			this.localizedDefaultStylesTitles );
+			plugins.get( 'ImageStyleEditing' ).normalizedStyles,
+			this.localizedDefaultStylesTitles
+		);
 
 		for ( const styleConfig of definedStyles ) {
 			this._createButton( styleConfig );
 		}
 
-		const definedGroups = translateStyles(
-			normalizedStyles.groups,
-			this.localizedDefaultStylesTitles );
+		const definedDropdowns = translateStyles(
+			[ ...toolbarConfig.filter( isObject ), ...utils.getDefaultDropdowns( plugins ) ],
+			this.localizedDefaultStylesTitles
+		);
 
-		for ( const groupConfig of definedGroups ) {
-			this._createDropdown( groupConfig, definedStyles );
+		for ( const dropdownConfig of definedDropdowns ) {
+			this._createDropdown( dropdownConfig, definedStyles );
 		}
 	}
 
 	/**
 	 * Creates a dropdown and stores it in the editor {@link module:ui/componentfactory~ComponentFactory}.
 	 *
 	 * @private
-	 * @param {module:image/imagestyle~ImageStyleGroupDefinition} dropdownConfig
+	 * @param {module:image/imagestyle~ImageStyleDropdownDefinition} dropdownConfig
 	 * @param {Array.<module:image/imagestyle~ImageStyleOptionDefinition>} definedStyles
 	 */
 	_createDropdown( dropdownConfig, definedStyles ) {
 		const factory = this.editor.ui.componentFactory;
 
-		factory.add( getUIComponentName( dropdownConfig.name ), locale => {
-			const { defaultItem, items, title, icon } = dropdownConfig;
+		factory.add( dropdownConfig.name, locale => {
+			let defaultButton;
+
+			const { defaultItem, items, title } = dropdownConfig;
 			const buttonViews = items
-				.filter( itemName => definedStyles.find( definedStyles => definedStyles.name === itemName ) )
-				.map( buttonName => factory.create( getUIComponentName( buttonName ) ) );
+				.filter( itemName => definedStyles.find( ( { name } ) => getUIComponentName( name ) === itemName ) )
+				.map( buttonName => {
+					const button = factory.create( buttonName );
+
+					if ( buttonName === defaultItem ) {
+						defaultButton = button;
+					}
+
+					return button;
+				} );
 
-			if ( !buttonViews.length || dropdownConfig.items.length !== buttonViews.length ) {
-				utils.warnInvalidStyle( { group: dropdownConfig } );
+			if ( items.length !== buttonViews.length ) {
+				utils.warnInvalidStyle( { dropdown: dropdownConfig } );
 			}
 
 			const dropdownView = createDropdown( locale, SplitButtonView );
 			const splitButtonView = dropdownView.buttonView;
-			const defaultIcon = definedStyles.find( item => item.name === defaultItem ).icon;
 
 			addToolbarToDropdown( dropdownView, buttonViews );
 
 			splitButtonView.set( {
-				label: title,
-				icon,
+				label: getDropdownButtonTitle( title, defaultButton.label ),
 				class: null,
 				tooltip: true
 			} );
 
 			splitButtonView.bind( 'icon' ).toMany( buttonViews, 'isOn', ( ...areOn ) => {
-				const index = areOn.findIndex( isOn => isOn );
+				const index = areOn.findIndex( identity );
```

Comment:
It took me a while to figure out what it all means but I guess this is a nice simplification if we get used to it 👍